pkg/kubectl/cmd/util/clientcache.go

```diff
@@ -46,14 +49,41 @@ func (c *clientCache) ClientConfigForVersion(version string) (*client.Config, er
 	}
 	// TODO: have a better config copy method
 	config := *c.defaultConfig
-	if len(version) != 0 {
-		config.Version = version
+	version, err := c.negotiateVersion(&config, version)
+	if err != nil {
+		return nil, err
 	}
+	config.Version = version
 	client.SetKubernetesDefaults(&config)
 
 	return &config, nil
 }
 
+func (c *clientCache) negotiateVersion(config *client.Config, version string) (string, error) {
+	client, err := client.New(config)
+	if err != nil {
+		return "", err
+	}
+	serverVersions, err := client.ServerAPIVersions()
+	if err != nil {
+		return "", fmt.Errorf("couldn't read version from server: %v\n", err)
+	}
+	for _, serverVersion := range serverVersions.Versions {
+		if version == serverVersion {
+			return version, nil
+		}
+	}
+	for _, clientVersion := range registered.RegisteredVersions {
```

Comment:
We probably need to make client cache agnostic to global static Kube client versions and instead let it be passed in.  I'll do that in a follow up (started this here https://github.com/smarterclayton/origin/commit/2d3aaa812a7dc6f25f8dfb31374a1da916054bf4) that can be post 1.0.